impacket/examples/secretsdump.py

```diff
@@ -1686,6 +1802,30 @@ def dump(self):
 
             self.__itemsFound[rid] = answer
             self.__perSecretCallback(answer)
+
+            if self.__history:
+                try:
+                    history = self.__extract_local_history(rid, newStyle, userAccount)
+                    lm_hist = history.get('lm', [])
+                    nt_hist = history.get('nt', [])
+                    while lm_hist and nt_hist and lm_hist[-1] == nt_hist[-1]:
+                        lm_hist.pop()
+                        nt_hist.pop()
+                    LOG.debug('History lengths for %s (RID %d): lm=%d nt=%d', userName, rid,
+                              len(lm_hist), len(nt_hist))
+                    count = max(len(lm_hist), len(nt_hist))
+                    for idx in range(count):
+                        lm_val = lm_hist[idx] if idx < len(lm_hist) else b''
+                        nt_val = nt_hist[idx] if idx < len(nt_hist) else b''
+                        lm_hex = hexlify(lm_val).decode('utf-8') if lm_val else empty_lm_hex
+                        nt_hex = hexlify(nt_val).decode('utf-8') if nt_val else empty_nt_hex
+                        if lm_hex == empty_lm_hex and nt_hex == empty_nt_hex:
+                            continue
+                        history_line = f"{userName}_history{idx}:{rid}:{lm_hex}:{nt_hex}:::"
+                        self.__historyItems.append(history_line)
+                        self.__perSecretCallback(history_line)
+                except Exception as exc:
+                    LOG.debug('SAM history parsing failed for RID %d: %s', rid, exc, exc_info=True)
```

Comment:
Shouldn't this be better as error?

```suggestion
                    LOG.error('SAM history parsing failed for RID %d: %s', rid, exc, exc_info=True)
```